Plugins/BuildServerIntegration/AzureDevOpsIntegration/ApiClient.cs

```diff
@@ -118,9 +118,16 @@ public async Task<IEnumerable<Build>> QueryBuildsAsync(string buildDefinitionsTo
 
             var builds = (await HttpGetAsync<ListWrapper<Build>>($"build/builds?api-version=2.0&definitions={buildDefinitionsToQuery}")).Value;
 
+            if (!running.HasValue || running.Value)
+            {
+                return builds
+                    .Where(b => !running.HasValue || running.Value == b.IsInProgress)
+                    .Where(b => !sinceDate.HasValue || b.StartTime >= sinceDate.Value.ToUniversalTime())
+                    .ToList();
```

Comment:
Do we need to materialise the collection here? The caller will be re-materialising them again, won't they?

Reply:
FIxed.